hedera-node/docs/design/services/token-service/airdrops/airdrop-transaction.md

```diff
@@ -0,0 +1,206 @@
+
+If the sender of the airdrop is also the payer of this transaction, we would need only their key to be present in the transaction context.
+
+If the airdrop transaction fees from the list above are covered by a separate payer, then the signatures of both the airdrop sender and the payer must be present in the context. This is so, because the sender is required to authorize sending the airdrop tokens from their balance.
+
+### Service updates
+
+- Update `TokenServiceDefinition` class to include the new RPC method definition for sending airdrops
+- Implement new `TokenAirdropHandler` class which should be invoked when the gRPC server handles `TokenAirdropTransaction` transactions. The class should be responsible for:
+    - Verify that the pending airdrops list contains between 1 and 10 entries, inclusive
+    - Pre-handle:
+        - The transaction must be signed by the sender for each entry in the `token_transfers` list
+        - In the case where the airdrop fees are covered by a separate payer, the transaction must include also the payer’s signature
+    - Handle:
+        - Any additional validation depending on config or state i.e. semantics checks
+        - Check that the airdrop sender account is a valid account. That is an existing account, which is not deleted or expired.
+        - Check that the token being airdropped, does not contain a custom fee, which needs to be covered by the recipient. This includes `fractionalFees` with `net_of_transfers=false` and `royaltyFees`, including `fallbackRoyaltyFees`.
+        - Such transactions should be reverted and rejected.
+        - The business logic for sending pending airdrops
+            - Check the association status and the existence of the recipient:
+              - It is an existing one and has `max_auto_associations` set to -1 and is not associated to the token →<br/>
+                Тhe token is auto associated and is directly transferred to the recipient and a `TokenTransferList` is added to the `TransactionRecord`, as well as a new `TokenAssociation` entry in the `automatic_token_associations`
+              - It is an existing one and has `max_auto_associations`, which is a positive number and there are free auto association slots and the recipient is not associated to the token →<br/>
+                Тhe token is auto associated and is directly transferred to the recipient and a `TokenTransferList` is added to the `TransactionRecord`, as well as a new `TokenAssociation` entry in the `automatic_token_associations`
+              - It is an existing one and has `max_auto_associations`, which is a positive number but there are no free auto association slots and the recipient is not associated to the token →<br/>
+                Тhe token transfer is interpreted as a pending and a `TokenPendingAirdrop` entry is added to the `TransactionRecord`. In addition, the airdrop `PendingAirdropId`/`PendingAirdropValue` entry is added to the `PendingAirdrop` VirtualMap in state
```

Comment:
`VirtualMap` is one possible implementation. In the business logic layer, we should only reference stores.

Reply:
Fixed